extensions/dicom-pdf/src/DicomPDFViewport.js

```diff
@@ -1,23 +1,135 @@
 import React, { Component } from 'react';
 import dicomParser from 'dicom-parser';
+import PDFJS from 'pdfjs-dist';
+import PropTypes from 'prop-types';
+
 import TypedArrayProp from './TypedArrayProp';
+import './DicomPDFViewport.css';
+
+import pdfjsBuild from 'pdfjs-dist/build/pdf';
+import pdfjsWorker from 'pdfjs-dist/build/pdf.worker.entry';
+pdfjsBuild.GlobalWorkerOptions.workerSrc = pdfjsWorker;
 
 // TODO: Should probably use dcmjs for this
 const SOP_CLASS_UIDS = {
   ENCAPSULATED_PDF: '1.2.840.10008.5.1.4.1.1.104.1',
 };
 
 class DicomPDFViewport extends Component {
-  state = {
-    fileURL: null,
-    error: null,
-  };
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      [this.props.viewportIndex]: {
+        fileURL: null,
+        error: null,
+        currentPageIndex: 1,
+        pdf: null,
+        scale: 1,
+        canvas: null,
+      },
+    };
+  }
 
   static propTypes = {
     byteArray: TypedArrayProp.uint8,
+    useNative: PropTypes.bool,
```

Comment:
Ah! This is clever. If we're going to `useNative` to toggle behavior, we could also code-split the `pdfjs` imports in mounted or updated? Then, if we're using the `useNative` prop, we never pull in pdfjs